wellcomeml/ml/bert_semantic_equivalence.py

```diff
@@ -170,30 +179,27 @@ def fit(self, X, y, random_state=None, epochs=3, metrics=[], **kwargs):
             )
 
             # Generates tensorflow dataset
-            self.train_dataset = self._prep_dataset_generator(X_train, y_train)
-            self.valid_dataset = self._prep_dataset_generator(X_valid, y_valid)
-
-            # Generates mini-batches and stores in a class variable
-            self.train_dataset = (
-                self.train_dataset.shuffle(self.max_length)
-                .batch(self.batch_size)
-                .repeat(-1)
-            )
-            self.valid_dataset = self.valid_dataset.batch(self.eval_batch_size)
+
```

Comment:
Note that prior to that you do consume a lot of memory when you split to train and val but I am not sure how best to avoid that without passing these pre splitted.